### ISSUE TITLE
Add support for sample-splitting  with validation set

### ISSUE THREAD

**Issue by JSzitas** [JSzitas/postDoubleR#3]
The original paper enables hyperparameter validation - lets try to support that too.
